caluma/caluma_workflow/tests/test_reopen_case.py

```diff
@@ -0,0 +1,214 @@
+import pytest
+from django.core.exceptions import ValidationError
+from graphene_django.utils.str_converters import to_const
+
+from .. import api, models
+
+
+def reopen_case_setup(case_status, work_item_statuses, case_factory, work_item_factory):
+    case = case_factory(status=case_status)
+    work_items = []
+
+    for work_item_status in work_item_statuses:
+        work_items.append(work_item_factory(case=case, status=work_item_status))
+
+    return case, work_items
+
+
+@pytest.mark.parametrize("use_graphql", [False, True])
+def test_allow_completed_cases_to_be_reopened(
+    db, admin_user, case_factory, work_item_factory, schema_executor, use_graphql
+):
+    completed_case, completed_work_items = reopen_case_setup(
+        models.Case.STATUS_COMPLETED,
+        [models.WorkItem.STATUS_COMPLETED],
+        case_factory,
+        work_item_factory,
+    )
```

Comment:
This test has exactly one statement common between the parametrisations, everything else is in separate code paths. Therefore I suggest splitting this test into two (once with graphql, once with api)